src/derivative_wrappers.jl

```diff
@@ -42,7 +42,8 @@ jacobian_autodiff(f, x, odefun, alg) = (ForwardDiff.derivative(f,x),1, alg)
 function jacobian_autodiff(f, x::AbstractArray, odefun, alg)
   sparsity = odefun.sparsity
   jac_prototype = odefun.jac_prototype
-  colorvec = DiffEqBase.has_colorvec(odefun) ? odefun.colorvec : (isnothing(sparsity) ? 1:length(x) : matrix_colors(sparsity))
+  colorvec = DiffEqBase.has_colorvec(odefun) ? odefun.colorvec :
+                  (isnothing(sparsity) ? (1:length(x)) : matrix_colors(sparsity))
```

Comment:
Maybe
```suggestion
  colorvec = DiffEqBase.has_colorvec(odefun) ? odefun.colorvec :
                  (sparsity === nothing ? (1:length(x)) : matrix_colors(sparsity))
```
just to make sure that it's easy for the compiler (although I doubt that it makes a difference here)?

Reply:
doesn't `isnothing` do `===` ?

Reply:
The result is the same, but it is easier for the compiler to reason about type inference with `===` than with the function call `isnothing`, see, e.g., https://github.com/JuliaLang/julia/issues/35585#issuecomment-619053871. I guess it's much more important though in, e.g., iteration.